### PR TITLE
Documentation for diffusion-reaction demo

### PR DESCRIPTION
**Pull request by tcsmaster**
Documentation for the diffusion-reaction equation

### CODE REVIEW

**Comment by lululxvi** [comment]
Rename the file name as diffusion.reaction.rst

**Comment by tcsmaster** [comment]
If I want to add more documentation, should I keep it in this PR?

**Comment by lululxvi** [comment]
> If I want to add more documentation, should I keep it in this PR?

You can do a new PR.

File: docs/demos/pinn_forward.rst [hunk 1]
```diff
@@ -46,9 +46,9 @@ Time-dependent PDEs
    pinn_forward/burgers.rar
    pinn_forward/allen.cahn
    pinn_forward/klein.gordon
+   pinn_forward/diffusion_reaction
```
Comment (lululxvi) [inline docs/demos/pinn_forward.rst:49]: Wrong file name.

File: docs/demos/pinn_forward/diffusion.reaction.rst [hunk 2]
```diff
@@ -0,0 +1,181 @@
+===========================
```
Comment (lululxvi) [inline docs/demos/pinn_forward/diffusion.reaction.rst:1]: Delete line 1.

File: docs/demos/pinn_forward/diffusion.reaction.rst [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+If the backend is Pytorch, the residual should look like this:
+
+.. code-block:: python
+
+    def pde(x, y):
+        dy_t = dde.grad.jacobian(y, x, i=0, j=1)
+        dy_xx = dde.grad.hessian(y, x, i=0, j=0)
+        d = 1
+        # Backend tensorflow.compact.v1 or tensorflow
```
Comment (lululxvi) [inline docs/demos/pinn_forward/diffusion.reaction.rst:70]: Is this comment correct?

File: docs/demos/pinn_forward/diffusion.reaction.rst [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    geom = dde.geometry.Interval(-1, 1)
+    timedomain = dde.geometry.TimeDomain(0, 0.99)
+    geomtime = dde.geometry.GeometryXTime(geom, timedomain)
+
+Now, we have specified the geometry and the PDE residual. We then define the ``TimePDE`` problem as
+
+.. code-block:: python
+
+    data = dde.data.TimePDE(
+    geomtime, pde, [], num_domain=320, solution=func, num_test=80000
```
Comment (lululxvi) [inline docs/demos/pinn_forward/diffusion.reaction.rst:112]: Wrong format.

File: docs/demos/pinn_forward.rst [hunk 5]
```diff
@@ -46,9 +46,9 @@ Time-dependent PDEs
    pinn_forward/burgers.rar
    pinn_forward/allen.cahn
    pinn_forward/klein.gordon
+   pinn_forward/diffusion.reaction
```
Comment (lululxvi) [inline docs/demos/pinn_forward.rst:49]: Reorder
```
pinn_forward/diffusion.1d.resample
pinn_forward/diffusion.reaction
pinn_forward/burgers.rar
```

**Pull request merged**